Sources/Twift.swift

```diff
@@ -5,13 +5,28 @@ import Combine
 public class Twift: NSObject, ObservableObject {
   /// The type of authentication access for this Twift instance
   public private(set) var authenticationType: AuthenticationType
+  public var oauthUser: OAuth2User? {
+    switch authenticationType {
+      case .oauth2UserAuth(let user):
+        return user
+      default:
+        return nil
+    }
+  }
+  
+  /// Optional callback to be performed after a successful token refresh.
+  private var refreshCompletion: (()-> Void)? = nil
   
   internal let decoder: JSONDecoder
   internal let encoder: JSONEncoder
   
   /// Initialise an instance with the specified authentication type
-  public init(_ authenticationType: AuthenticationType) {
+  public init(_ authenticationType: AuthenticationType, onRefresh: (()-> Void)? = nil) {
```

Comment:
Seeing this now, it actually makes more sense to have `onRefresh` associated with the `AuthenticationType`:

```swift
enum AuthenticationType {
  case oauth2User(user: OAuth2User, onRefresh: (() -> Void)?)
  ...
}
```

Alternatively, we might want to move to different initialisers for different authentication types. That might actually be nicer from an author perspective.

Reply:
I see, yeah, makes sense
Should we then even make it non-optional? I am not sure what the case would be where one would not need a refresh of the token?

I made some changes to the `refreshOAuth2AccessToken` and moved `refreshCompletion` from the init to the `AuthenticationType`